Cannot build on Arch Linux

These are all the variable, which are associated with `SDL2` are set in the CMake after `find_package(SDL2 REQUIRED)`:
```
$ cmake .. | grep SDL2
-- CONFIG_FILES=/usr/lib64/cmake/SDL2/SDL2Targets-noconfig.cmake
-- SDL2_CONFIG=/usr/lib64/cmake/SDL2/SDL2Config.cmake
-- SDL2_CONSIDERED_CONFIGS=/usr/lib64/cmake/SDL2/SDL2Config.cmake
-- SDL2_CONSIDERED_VERSIONS=2.0.9
-- SDL2_DIR=/usr/lib64/cmake/SDL2
-- SDL2_FOUND=1
-- SDL2_VERSION=2.0.9
-- SDL2_VERSION_COUNT=3
-- SDL2_VERSION_MAJOR=2
-- SDL2_VERSION_MINOR=0
-- SDL2_VERSION_PATCH=9
-- SDL2_VERSION_TWEAK=0
-- _DIR=/usr/lib64/cmake/SDL2
```

According to [this](https://discourse.libsdl.org/t/arch-linux-cmake-find-package-sdl2-required-passes-but-doesnt-find-anything/24226/2) all we have to do is to detect if build OS is arch, and add target link libraries `SDL2::SDL2`.

**Edit**: I've tested and it fixes the issue. Going to submit a PR soon.
**Edit 2**: I couldn't find an easy way of detecting the Arch Linux environment. So, for now it can be just mentioned in the docs that `SDL2::SDL2` should be added to `target_link_libraries` if building on Arch.

> why do we need to detect ArchLinux? Are the two build approaches so mutually incompatible? 
We can detect if SDL2_LIBRARIES is empty, and set it to `SDL2::SDL2` manually with the following code:
```
if("${SDL2_LIBRARIES}" STREQUAL "")
    message(WARNING "SDL2_LIBRARIES wasn't set, manually setting to SDL2::SDL2")
    set(SDL2_LIBRARIES "SDL2::SDL2")
endif()
```
See PR below.

The following error
```
CMake Error at CMakeLists.txt:16 (find_package):
  By not providing "FindSDL2.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "SDL2", but
  CMake did not find one.

  Could not find a package configuration file provided by "SDL2" with any of
  the following names:

    SDL2Config.cmake
    sdl2-config.cmake

  Add the installation prefix of "SDL2" to CMAKE_PREFIX_PATH or set
  "SDL2_DIR" to a directory containing one of the above files.  If "SDL2"
  provides a separate development package or SDK, be sure it has been
  installed.
```
occurs on my linux machine when I don't have SDL2 installed (the same error is in MacOS CI). When SDL2 is installed, `cmake` configuration completes successfully. But when I run build with `cmake --build .` in the `build` directory, I get the following error:
```
Scanning dependencies of target nothing
[ 33%] Building C object CMakeFiles/nothing.dir/src/broadcast.c.o
In file included from /home/richard/projects/c/nothing/src/broadcast.c:10:
/home/richard/projects/c/nothing/src/game.h:4:10: fatal error: SDL.h: No such file or directory
    4 | #include <SDL.h>
```
As I've already said above, this error resolves when I manually add `SDL2::SDL2` to the `target_link_libraries` (because no variables are set by `find_package` command).

So, I think these errors are not related.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,11 @@ else()
   find_package(SDL2 REQUIRED)
 endif()
 
+if("${SDL2_LIBRARIES}" STREQUAL "")
+    message(WARNING "SDL2_LIBRARIES wasn't set, manually setting to SDL2::SDL2")
+    set(SDL2_LIBRARIES "SDL2::SDL2")
+endif()
+
 include_directories(${CMAKE_BINARY_DIR})
 include_directories(src/)
 

